docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+
+Our Lit Action primarily focuses on three factors:
+
+- Access Conditions for checking funds on both chains
+- Transaction objects for transferring funds between swap parties
+- Clawback Transaction objects for revert transferring funds to their owners
+- Lit Action conditions which decide which transaction object to sign and return
+
+We'll use [Lit's Access Control](../sdk/access-control/evm/basic-examples#must-posess-at-least-one-erc20-token) here to verify if the conditions are being met for the swap. There need to be 2 conditions for a swap to execute,
+
+- Have token A funds reached on Chain A?
+- Have token B funds reached on Chain B?
+
+If yes, then forward the funds and complete the swap or else revert them.
+
+A condition on Base Sepolia for checking if the PKP contains the exact amount of token A in Wei will look like this:
+
+```js
+const chainACondition = {
+    conditionType: "evmBasic",
+    contractAddress: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    standardContractType: "ERC20",
+    chain: "baseSepolia",
+    method: "balanceOf",
+    parameters: ["address"],
+    returnValueTest: { comparator: "==", value: "4000000000000000000" },
+};
+// parameter field would be later replaced by the pkpAddress in the action
+```
+
+There need to be two transaction objects, each of which transfers funds between the swap parties. Each of them will be calling function `transfer(address, uint256)` on the token's smart contract. This information is captured in the "data" field of the object. We can get that in the following way,
+
+```js
+function generateCallData(counterParty, amount) {
+    const transferInterface = new ethers.utils.Interface([
+        "function transfer(address, uint256) returns (bool)",
+    ]);
+    return transferInterface.encodeFunctionData("transfer", [
+        counterParty,
+        amount,
+    ]);
+}
+```
+
+A transaction object would then look something like below
+
+```js
+let chainATransaction = {
+    to: "0xad50f302a957C165d865eD398fC3ca5A5A2cDA85",
+    gasLimit: "60000",
+    from: "0x291B0E3aA139b2bC9Ebd92168575b5c6bAD5236C",
+    data: "0xa9059cbb000000000000000000000000291b0e3aa139b2bc9ebd92168575b5c6bad5236c0000000000000000000000000000000000000000000000003782dace9d900000",
+    type: 2,
+};
+// "to" field specifies tokenA contract
+// "from" field would be later replaced by the pkpAddress in the action
+```
+
+Clawback transaction objects are transactions that transfer funds back to the depositors if the swap conditions aren't met. These will have a different value for the `data` field and can be generated again using the `generateCallData()` function, with the `counterParty` value set to the depositor.
+
+Let's start with writing a Lit Action,
```

Comment:
Maybe something like "Now that we've covered all of the relevant background information, let's write our Lit Action:"